### PR TITLE
Add new `batch_format` method.

### LINKED ISSUE #929: add batch_format method to Worksheet eliminating the need for multi request for formatting cells
**Is your feature request related to a problem? Please describe.**
i recently made a small program with this package. it is a great experience!
the issue came when i needed to format some cells in arbitrary places (not in a range).
that forced me to make lots of `Worksheet.format` calls (which results in an api fired to google). that method proved ok until my the formatting need exceeded my quota (or other network/time constrains).

i eventually made a helper function and class for storing format requests and then fired them with `Spreadsheet.batch_update`.

i would love to share my solution via PR if it would also aid anyone here (`CONTRIBUTING.md` asked to put an issue first))

**Describe the solution you'd like**
i would like a solution that stores format requests and then be able to fire all of them at once (with a single api call to google).

**Describe alternatives you've considered**
simply wait until i can make another api calls to google and them continue the program (this is BAD)

i have already made the solution to my project, would love to share it.

### PR DESCRIPTION
**Pull request by lavigne958**
This method allows end-user to format multiple non-overlapping ranges in
a single API call.

Updated method `format` to accept a `str` for a single range or a `list`
of `str` to apply the same format to multiple non overlapping ranges.

Closes #929

### CODE REVIEW

**Comment by ericbn** [comment]
Cool! I was working on creating a PR for this (and you did a better job than mine!), but was unable to record a new cassette for the new test case I wrote. I was getting:

```
self = <tests.conftest.SleepyClient object at 0x10dba1c10>, method = 'post', endpoint = 'https://sheets.googleapis.com/v4/spreadsheets/1f6e0m42XXAsEMJnr5-lzOf6srAjvBmOuuQlmU0npwR4/values/%27Sheet1%27:clear', params = None, data = None, json = None, files = None
headers = None

...
>           raise APIError(response)
E           gspread.exceptions.APIError: {'code': 404, 'message': 'Requested entity was not found.', 'status': 'NOT_FOUND'}
```
Maybe the original spreadsheet used for the tests is not available anymore...

File: gspread/worksheet.py [hunk 1]
```diff
@@ -919,20 +919,87 @@ def batch_update(self, data, **kwargs):
...
+                    }
+                }
+            )
+
+        return self.spreadsheet.batch_update(body)
+
+    def format(self, ranges, format):
+        """Format a list of ranges with the given format.
+
+        :param str|list ranges: Target ranges in the A1 notation.
```
Comment (ericbn) [inline gspread/worksheet.py:991]: Didn't know this was possible:
```
:param str|list ranges: ...
```

I thought it had to be separated into:
```
:param ranges: ...
:type ranges: str or list
```

**Comment by lavigne958** [comment]
> Cool! I was working on creating a PR for this (and you did a better job than mine!), but was unable to record a new cassette for the new test case I wrote. I was getting:
> 
> ```
> self = <tests.conftest.SleepyClient object at 0x10dba1c10>, method = 'post', endpoint = 'https://sheets.googleapis.com/v4/spreadsheets/1f6e0m42XXAsEMJnr5-lzOf6srAjvBmOuuQlmU0npwR4/values/%27Sheet1%27:clear', params = None, data = None, json = None, files = None
> headers = None
> 
> ...
> >           raise APIError(response)
> E           gspread.exceptions.APIError: {'code': 404, 'message': 'Requested entity was not found.', 'status': 'NOT_FOUND'}
> ```
> Maybe the original spreadsheet used for the tests is not available anymore...

Hi sorry to hear that. I had no response from you when I asked you if you wanted to do it so after 3 days I needed to do something to keep deliveries on time.


About the recording you faced the following issue:
- you configure the test suite to record the new test
- the suits starts by opening the spreadsheet
- it gets the result from the recorded cassette
- then you make the real request to the API because you don't have the record cassette for your new test.
- boom it fails, because the spreadsheet does not exists for your current test online.

How to solve this:
- delete the record cassette tight to your test.
- start your test again with the record mode to `new_episodes` it will record everything and should pass.

File: gspread/worksheet.py (hunk 1 above)
Comment (lavigne958) [reply gspread/worksheet.py:991]: To be honest I tried multiple options this one worked i yav was quiet happy 😁

**Comment by ericbn** [comment]
No worries! I should have responded earlier. And thanks for the heads up about solving the cassette recoding issue! Will try it next time.

**Pull request merged**